cidc_api/models/models.py

```diff
@@ -345,6 +349,54 @@ def disable_inactive_users(session: Session, commit: bool = True):
             session.commit()
         return res
 
+    @staticmethod
+    @with_default_session
+    def get_data_access_report(io: BinaryIO, session: Session) -> pd.DataFrame:
+        """
+        Generate an XLSX containing an overview of trial/assay data access permissions
+        for every active user in the database. The report will have a sheet per protocol
+        identifier, with each sheet containing columns corresponding to a user's email,
+        organization, role, and upload type access permissions.
+
+        Save an excel file to the given file handler, and return the pandas dataframe
+        used to generate that excel file.
+        """
+        user_columns = (Users.email, Users.organization, Users.role)
+
+        query = (
+            session.query(
+                *user_columns,
+                Permissions.trial_id,
+                func.string_agg(Permissions.upload_type, ","),
```

Comment:
nit: i'd go with `array_agg` and do formatting in python instead

Reply:
Ah, sure - I think I'm going to forgo this for now until there's indication we need the permissions info in a more flexible format.